pkg/daemon/discover/discover.go

```diff
@@ -211,16 +211,17 @@ func udevBlockMonitor(c chan string, period time.Duration) {
 		[]string{"(?i)add", "(?i)remove"},
 		udevFilter)
 
+	timeout := time.NewTimer(period)
+	defer timeout.Stop()
 	for {
-		event, ok := <-events
+		_, ok := <-events
 		if !ok {
 			return
 		}
-		timeout := time.NewTimer(period)
+		timeout.Reset(period)
```

Comment:
Do we also need to drain the `timeout.C` channel in case the previous timer fired?

Reply:
I don't think it's necessary
1. `t.C` will be drained in the second for loop even if events reach.
2. There is no race condition, because timer is used in the same goroutine.

Reply:
According to [docs](https://pkg.go.dev/time#Timer.Reset), usage of `Reset()` in this context is wrong. 

The reset should be done as so
```go
if !t.Stop() {
	<-t.C
}
t.Reset(d)
```